libvast/builtins/endpoints/query.cpp

```diff
@@ -206,13 +207,14 @@ struct query_manager_state {
   http_request request;
   size_t limit = 0u;
   std::string response_body; // The current response to the GET endpoint
-  std::deque<table_slice> slice_buffer;
-  // Events produced after application of a pipeline.
-  std::deque<table_slice> processed_slices;
+  std::deque<table_slice> source_buffer;
+  std::deque<table_slice> sink_buffer;
   size_t shippable_events_count = 0;
   std::optional<system::query_cursor> cursor = std::nullopt;
   size_t processed_partitions = 0u;
-  std::optional<pipeline_executor> pipeline_executor_;
+  bool active_index_query = false;
+  generator<caf::expected<void>> executor;
+  generator<caf::expected<void>>::iterator executor_it = {};
```

Comment:
I like this idea. Alternatively, we could also add a method to `generator` that just returns `iterator{m_coroutine}` (although this is only safe to call if `.begin()` was called before).

Reply:
I would prefer for this to have a separate name but to be constructible from a generator. The usage is very different, which to me implies that it should have a different name.

Reply:
The alternative suggestion would not add a `next()` method, but would just provide a way to get the current `generator<caf::expected<void>>::iterator` without storing it anywhere (because `executor_it` is basically a constant, i.e., `iterator{m_coroutine}`). That's the same usage, just without storing the iterator anywhere.

But we can also add a generator with `auto next() -> std::optional<T>`, that would also be useful.

Reply:
Should we move this discussion into a follow-up issue? I don't think it should be blocking this from being merged.